backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/InterestComputationCommand.java

```diff
@@ -371,37 +371,37 @@ private void distributeInterest(
 	}
 
 	@NonNull
-	private InterimContractAllocations initInterimContractAllocations(
+	private ModularContractAllocations initModularContractAllocations(
 			@NonNull final InterestComputationRequest request,
 			final int additionalInterestDays,
-			@NonNull final ModularContractLogEntry interimContractEntry)
+			@NonNull final ModularContractLogEntry modularContractLogEntry)
 	{
-		Check.assume(LogEntryDocumentType.CONTRACT_PREFINANCING == interimContractEntry.getDocumentType(),
-					 "Expecting DocumentType = " + LogEntryDocumentType.CONTRACT_PREFINANCING +
-							 " but got " + interimContractEntry.getDocumentType() + "! LogId=" + interimContractEntry.getId());
-		Check.assumeNotNull(interimContractEntry.getAmount(), "Invoices with no amount should've been skipped already! LogId="
-				+ interimContractEntry.getId());
+		Check.assume(LogEntryDocumentType.PURCHASE_MODULAR_CONTRACT == modularContractLogEntry.getDocumentType(),
+					 "Expecting DocumentType = " + LogEntryDocumentType.PURCHASE_MODULAR_CONTRACT +
+							 " but got " + modularContractLogEntry.getDocumentType() + "! LogId=" + modularContractLogEntry.getId());
+		Check.assumeNotNull(modularContractLogEntry.getAmount(), "Invoices with no amount should've been skipped already! LogId="
```

Comment:
```suggestion
		Check.assumeNotNull(modularContractLogEntry.getAmount(), "Contracts with no amount should've been skipped already! LogId="
```

I'm not even sure this check makes sense, because if the contract is for amount 0, I question their business model.